### DISCUSSION TITLE
Возможно ли это реализовать на Totum?

### DISCUSSION THREAD

**Discussion by rusovich-sergey** [totumonline/totum-ru-issues-and-discussions#366] [Как это сделать?] [answered]
Появилась необходимость организовать автоматизацию заказов товара на розничных магазинах.
На сегодняшний день работаем с гугл доками, менеджер каждую неделю загружает прайс листы в виде электронной таблицы в гугл, отдельный документ на отдельный бренд, документ содержит наименование товара, остаток и графу для проставления количества заказа. Непосредственно на магазине продавец открывает этот документ, проставляет количество и отправляет заполненный документ по электронной почте. Менеджер на офисе получает документ и обрабатывает его (вносит в 1С и отдает на сборку).
Организация выросла, стало много магазинов и брендов, с гугл доками стало очень неудобно работать. Можно ли что-то похожее организовать на Totum? 
Например:
Администратор создает загрузкой (xml или xlsx или ещё какой формат) в Totum'е таблицы, каждый бренд своя таблица.
Менеджер обновляет остатки импортом из 1С.
Продавец (каждый под своим логином) видит все таблицы с брендами, при открытии любой из таблиц видит товар только с положительным остатком и может проставить количество для заказа, после сохранения таблицы, создается XLSX с отметкой "оформленный заказ".
Менеджер видит все новые заказы, скачивает созданный XLSX документ, после чего на этом документе ставится пометка "ушел в обработку".
Реально ли реализовать такую задачу на Totum? Если да, достаточно ли будет пройти учебный курс для самостоятельной реализации ?

**Comment by erpic-ru** [comment, marked as answer]
Да

Пара примечаний:
1. Таблица может быть одной с префильтром по брендам
2. Непонятно зачем в этой схеме нужен эксель и почта, тк менеджер может получить push и открыть документ в Тотуме. В принципе можно и прямой обмен с 1С настроить в дальнейшем

**Reply by ttm-alexeypolunin** [reply]
Мне тоже кажется, что эксель здесь лишний — у вас в онлайне может быть таблица с остатками и магазины в системе будут размещать заказы.

Если не было опыта разработки, то лимит сложности, который у вас получится самостоятельно будет ниже среднего, но главное тут за 2-3 месяца перевести в эксплуатацию.

Можете записаться на консультацию (20 мин я объясню про архитектуру): https://control.demoru.totum.online/Forms/free-consultation

**Reply by rusovich-sergey** [reply]
> Да
> 
> Пара замечаний:
> 
> 1. Таблица может быть одной с префильтром по бренду
> 2. Непонятно, зачем в этой структуре нужен эксель и почта, т.к. менеджер может получить push и открыть документ в Тотуме. В обязательном порядке можно и прямо обмениваться с 1С повышать в дальнейшем

1. Если таблица будет одна, то в ней будет около 10К строк (не тяжело ли будет работать с такой таблицей ?), так же менеджеру иногда нужно будет забирать заказ только по одному бренду, а завтра по другому, нужно будет как-то отмечать какие заказы он забрал, что б не держать это в голове.
2. Почта уже не нужна будет, если будем работать в Totum'е (почту используем сегодня, при работе с гугл доками). Excel нужен на первое время, пока не настроен автоматический обмен с 1С, на сегодняшний день мы загружаем excel документы в 1С обработкой.

**Reply by ttm-alexeypolunin** [reply]
10М или 100М — в таких случаях можно подумать об оптимизации, а 10К может быть самым топорным способом (я видел как люди 40К выводили на один! экран в тотуме, тормозило адски и клиентский комп дымился, но работало).

И это таблица с итемами и остатками — тип Простая.

Я бы еще сделал табличку с движением — на основе которой изменяется остаток и для Заказа возможно использовал бы Циклы. Те магаз создает заявку и набирает, что ему туда нужно (тут 140 вариантов может быть как это сделано и вот здесь уже по конкретному бренду может нужно тк таблицы в цикле не бесконечные).

> А могут быть и бесконечные через табы в Циклах — тогда и заказ может быть на 10К строк от 100 магазинов параллельно (но я бы в заказ записывал только те строки, которые заказываются и не писал нулевые).

Еще неизвестно, что там с резервированием (остаток 1, а два магаза хотят доставку к себе).

**Reply by erpic-ru** [reply]
Ну это само-собой, что оптимальнее делать заявки в циклах, в которую добавляются строки из основной таблицы с номенклатурой, с контролем резерва и остатков. Но, возможно, стартануть можно и с 1 таблицы :)

**Reply by ttm-alexeypolunin** [reply]
Тогда простой вариант:

Табл 1 — остатки
Табл 2 — заказ
Табл 3 — движение

**Reply by rusovich-sergey** [reply]
> Тогда простой вариант:
> 
> Табл 1 — остатки Табл 2 — заказ Табл 3 — движение

Пока можно без контроля остатков, контроль реализован на стороне 1С, на тотуме будет сложнее реализовать т.к. этот товар могут двигать не только магазины но и оптовые покупатели. Достаточно будет что менеджер будет чаще синхронизировать остатки 1С и Totum. Если на остатках 1 шт, а её заказали два магазина, то достанется тому который будут раньше загружать в 1С.
